com.unity.render-pipelines.high-definition/Runtime/Lighting/ScreenSpaceLighting/SSGIDenoiser.cs

```diff
@@ -79,88 +77,236 @@ RTHandle IndirectDiffuseHistoryBufferAllocatorFunction(string viewName, int fram
+            var info = m_SharedRTManager.GetDepthBufferMipChainInfo();
+            parameters.firstMipOffset.Set(HDShadowUtils.Asfloat((uint)info.mipLevelOffsets[1].x), HDShadowUtils.Asfloat((uint)info.mipLevelOffsets[1].y));
+            parameters.historyValidity = historyValidity;
+            parameters.viewCount = hdCamera.viewCount;
+
+            // Denoising parameters
+            parameters.filterRadius = giSettings.filterRadius;
+            parameters.halfResolution = halfResolution;
+            parameters.historyValidity = historyValidity;
+            parameters.historyNeedsClear = historyNeedsClear;
+
+            // Compute shader
+            parameters.ssgiDenoiserCS = m_SSGIDenoiserCS;
+
+            // Kernels
+            parameters.spatialFilterKernel = halfResolution ? m_SpatialFilterHalfKernel : m_SpatialFilterKernel;
+            parameters.temporalFilterKernel = halfResolution ? m_TemporalFilterHalfKernel : m_TemporalFilterKernel;
+            parameters.copyHistory = m_CopyHistory;
+
+            return parameters;
+        }
+
+        struct SSGIDenoiserResources
+        {
+            // Input Buffers
+            public RTHandle depthTexture;
+            public RTHandle normalBuffer;
+            public RTHandle motionVectorsBuffer;
+
+            // History Buffer
+            public RTHandle indirectDiffuseHistory;
+            public RTHandle historyDepthBuffer;
+
+            // Intermediate buffer
+            public RTHandle intermediateBuffer;
+
+            // In-output Buffer
+            public RTHandle inputOutputBuffer;
+        }
+
+        SSGIDenoiserResources PrepareSSGIDenoiserResources(RTHandle historyDepthBuffer, RTHandle indirectDiffuseHistory, RTHandle inputOutputBuffer, RTHandle intermediateBuffer)
+        {
+            SSGIDenoiserResources resources = new SSGIDenoiserResources();
+
+            // Input Buffers
+            resources.depthTexture = m_SharedRTManager.GetDepthTexture();
+            resources.normalBuffer = m_SharedRTManager.GetNormalBuffer();
+            resources.motionVectorsBuffer = m_SharedRTManager.GetMotionVectorsBuffer();
+
+            // History Buffer
+            resources.indirectDiffuseHistory = indirectDiffuseHistory;
+            resources.historyDepthBuffer = historyDepthBuffer;
+
+            // Intermediate buffer
+            resources.intermediateBuffer = intermediateBuffer;
 
-            // Pick the right kernel to use
-            int m_KernelFilter = halfResolution ? m_SpatialFilterHalfKernel : m_SpatialFilterKernel;
+            // In-output Buffer
+            resources.inputOutputBuffer = inputOutputBuffer;
+
+            return resources;
+        }
+
+        static void Denoise(CommandBuffer cmd, SSGIDenoiserParameters parameters, SSGIDenoiserResources resources)
+        {
+            if (resources.historyDepthBuffer == null)
```

Comment:
Can we do this test outside the funciton? If it fails it means we still need to run a render graph pass through the system and we'll also get an empty profiling marker.